docs/content/utilities/parser.mdx

```diff
@@ -0,0 +1,508 @@
+```python:title=extending_builtin_models.py
+from aws_lambda_powertools.utilities.parser import parse, BaseModel
+from aws_lambda_powertools.utilities.parser.models import EventBridgeModel
+
+from typing import List, Optional
+
+class OrderItem(BaseModel):
+    id: int
+    quantity: int
+    description: str
+
+class Order(BaseModel):
+    id: int
+    description: str
+    items: List[OrderItem]
+
+# highlight-start
+class OrderEventModel(EventBridgeModel):
+    detail: Order
+# highlight-end
+
+payload = {
+    "version": "0",
+    "id": "6a7e8feb-b491-4cf7-a9f1-bf3703467718",
+    "detail-type": "OrderPurchased",
+    "source": "OrderService",
+    "account": "111122223333",
+    "time": "2020-10-22T18:43:48Z",
+    "region": "us-west-1",
+    "resources": ["some_additional"],
+    "detail": { # highlight-line
+        "id": 10876546789,
+        "description": "My order",
+        "items": [
+            {
+                "id": 1015938732,
+                "quantity": 1,
+                "description": "item xpto"
+            }
+        ]
+    }
+}
+
+ret = parse(model=OrderEventModel, event=payload) # highlight-line
+
+assert ret.source == "OrderService"
+assert ret.detail.description == "My order"
+assert ret.detail_type == "OrderPurchased" # we rename it to snake_case since detail-type is an invalid name
+
+for order_item in ret.detail.items:
+    ...
+```
+
+**What's going on here, you might ask**:
+
+1. We imported our built-in model `EventBridgeModel` from the parser utility
+2. Defined how our `Order` should look like
+3. Defined how part of our EventBridge event should look like by overriding `detail` key within our `OrderEventModel`
+4. Parser parsed the original event against `OrderEventModel`
+
+## Parsing events
+
+You can parse inbound events using **event_parser** decorator, or the standalone `parse` function. Both are also able to parse either dictionary or JSON string as an input.
+
+### event_parser decorator
+
+Use the decorator for fail fast scenarios where you want your Lambda function to raise an exception in the event of a malformed payload.
+
+`event_parser` decorator will throw a `ValidationError` if your event cannot be parsed according to the model.
+
+```python=:title=event_parser_decorator.py
+from aws_lambda_powertools.utilities.parser import event_parser, BaseModel, ValidationError
+from aws_lambda_powertools.utilities.typing import LambdaContext
+import json
+
+class OrderItem(BaseModel):
+    id: int
+    quantity: int
+    description: str
+
+class Order(BaseModel):
+    id: int
+    description: str
+    items: List[OrderItem] # nesting models are supported
+    optional_field: Optional[str] # this field may or may not be available when parsing
+
+payload = {
+    "id": 10876546789,
+    "description": "My order",
+    "items": [
+        {
+            "id": 1015938732,
+            "quantity": 1,
+            "description": "item xpto"
+        }
+    ]
+}
+
+@event_parser(model=Order) # highlight-line
```

Comment:
what about an example of model + envelope together? that's the most powerful & easy to use combination.

Reply:
other than that, amazing job to document this feature and pydantic

Reply:
Made 3 improvements there to address your feedback ;)

- Expanded that section to include `event_parser` using envelope param
- Explicit call out that both `event_parser` and `parse` support envelope param
- Note to call out `event_parser` will replace the event with the parsed model, if successful, in case someone is trying to use it with other nested decorators